kroxylicious-kms-provider-aws-kms/src/test/java/io/kroxylicious/kms/provider/aws/kms/AwsV4SigningHttpRequestBuilderTest.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kms.provider.aws.kms;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.net.InetSocketAddress;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.time.Instant;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import com.sun.net.httpserver.Headers;
+import com.sun.net.httpserver.HttpExchange;
+import com.sun.net.httpserver.HttpHandler;
+import com.sun.net.httpserver.HttpServer;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class AwsV4SigningHttpRequestBuilderTest {
+
+    private static final YAMLFactory YAML_FACTORY = new YAMLFactory();
+    private static final ObjectMapper MAPPER = new ObjectMapper(YAML_FACTORY).registerModule(new JavaTimeModule());
+    private static final String ACCESS_KEY = "access";
+    private static final String SECRET_KEY = "secret";
+    private static final String REGION = "us-east-1";
+    private static final String SERVICE = "kms";
+
+    static Stream<Arguments> requestSigning() throws Exception {
+        try (var knownGoodYaml = AwsV4SigningHttpRequestBuilderTest.class.getResourceAsStream("/io/kroxylicious/kms/provider/aws/kms/known_good.yaml")) {
+            assertThat(knownGoodYaml).isNotNull();
+            var parser = YAML_FACTORY.createParser(knownGoodYaml);
+            List<TestDef> testDefs = MAPPER.readValues(parser, TestDef.class).readAll();
+            return testDefs.stream().map(td -> Arguments.of(td.testName(), td));
+        }
+    }
+
+    /**
+     * The test compares known-good signatures generated by Curl's AWS v4 signing
```

Comment:
I also considered AWS's CLI to generate the known-good rather than Curl.  Curl turned out slightly easier as you have direct control of the urls.